src/actions/Scroll.ts

```diff
@@ -49,8 +47,23 @@ class Scroll implements Action {
       await focusEditor(originalEditor);
     }
 
+    const decorationSelections = targets
+      .map((target) => target.selection)
+      .filter((selection) => {
+        const visibleRanges = selection.editor.visibleRanges;
+        const start = visibleRanges[0].start;
+        const end = visibleRanges[visibleRanges.length - 1].end;
+        // Don't show decorations for selections that are larger than the visible range
+        return (
+          selection.selection.start.isAfter(start) ||
+          selection.selection.end.isBefore(end) ||
```

Comment:
I believe you want to use line numbers here instead of just a raw before or after. This doesn't work correctly if the target is indented

Reply:
You are correct. Code updated.